elytron/ELY-2078-encryption-for-FilesystemSecurityRealm.adoc

```diff
@@ -0,0 +1,250 @@
+                        .setRoot(getRootPath())
+                        .setLevels(3)
+                        .setSecretKey(key)
+                        .build();
+```
+This will create a new file system with 3 levels for the identity file, encrypted with the Secret Key "key".
+
+In WildFly Core, the filesystem-realm will include new attributes to specify the credential store and the alias within the store that holds the secret key that should be used to enable encryption for the filesystem realm.
+
+*Attributes*
+
+`credential-store` - a reference to the credential store that contains the secret key used to encrypt and decrypt the filesystem-realm. This is an optional attribute.
+
+`secret-key` - an alias to the secret key used to encrypt and decrypt the filesystem-realm. If the credential-store attribute is specified, this attribute is required. Otherwise this is an optional attribute.
+
+To create an encrypted FileSystem the following commands would be used
+
+```bash
+/subsystem=elytron/secret-key-credential-store=mycredstore:add(path=propcredstore.cs, relative-to=jboss.server.config.dir, create=true, populate=true)
+```
+```bash
+/subsystem=elytron/filesystem-realm=fsRealm:add(path=fs-realm-users, relative-to=jboss.server.config.dir, credential-store=mycredstore, secret-key=key)
+```
+The filename is stored as the the principal (also know as the username) encoded in base32, once an identity is created with encryption enabled.
+For example the identity with the username `plainUser` will be saved in a file named `OBWGC2LOKVZWK4Q.xml`.
+The identity file being encrypted with a SecretKey that contains a bcrypt hashed password along with 5 attributes would look like the following.
+
+```XML
+<?xml version="1.0" ?>
+<identity xmlns="urn:elytron:1.0">
+    <credentials>
+        <password algorithm="bcrypt" format="enc_base64">RUxZAUMQo2svngST8+WfAg6BoeBtBH0exXBlwAQvN4jCxEuDwPUkniZkgx/3mfetwDJK4rql0jfaigStMAW8mE9iNXG5QQ==</password>
+    </credentials>
+    <attributes>
+        <attribute name="RUxZAUMQUO9rq4PRc3BB61B5oZSepATonjzj1dlUriqbsy//5fw=" value="RUxZAUMQwbcAyojI0gsU0BFxeuBw/zJXR8Umy/jk9tnWrvpRxu0="></attribute>
+        <attribute name="RUxZAUMQ5iZ43cvM0x/RwPxwnqxXZisDmFX0Un+f3pLQz0WDC3o=" value="RUxZAUMQWUAYVdzpdv4sKGaFblcIMsso6x0j5J9oCSHQiDbjJE8="></attribute>
+        <attribute name="RUxZAUMQl1xAhDoJdsMnk7ggw66zyClhUzzB79p859BkQrXazzo=" value="RUxZAUMQCV32L5vunNfLMCISdjyn7myJ0saJdAWreKq1BjnGjfU="></attribute>
+        <attribute name="RUxZAUMQ7t3uOr63lNMwD+Ljsq68ONPwGYdtbVLtdmbY6stZIjg=" value="RUxZAUMQ41IKUmpazIiXd4wQa67I+y35TLtssweZ/1aci7w71ag="></attribute>
+        <attribute name="RUxZAUMQDIjiDvRMr9sOCHnCyh8/uwytBJZOfgCsH0k0itNwcNA=" value="RUxZAUMQi3C21O9+je0b9kXHIsQyoiuSYYafYAFKS3h7iPM9+OE="></attribute>
+    </attributes></identity>
+```
+=== Integrity of Filesystem Realm
+The two approaches being considered are between signing each identity file, and just calculating the checksum of each identity file.
```

Comment:
Same here, we should mention that is something that is being planned for after 26.1.

Reply:
Added